src/Table/Table.js

```diff
@@ -142,6 +142,14 @@ class Table extends Component {
     allRowsSelected: this.props.allRowsSelected,
   };
 
+  componentWillReceiveProps(nextProps) {
+    if (typeof nextProps.allRowsSelected !== 'undefined' && this.props.allRowsSelected !== nextProps.allRowsSelected) {
```

Comment:
Here we are allowing for the maintaining of allRowsSelected from outside of the component. This is necessary if utilizing "selected" on the data.

When maintaining the "selected" state outside of the table, you should expect to have to maintain the "all selected" state as well I believe.